### PR TITLE
Fixed updates for `databricks_share` resource

### LINKED ISSUE #2167: [ISSUE] Issue with `databricks_share` update detection
### Configuration
Adding a new `object` to a `databricks_share` resource e.g.
```hcl
databricks_share my_share {
  name = "my_share"
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_A"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_C"
  }
}
```
->
```hcl
databricks_share my_share {
  name = "my_share"
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_A"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_B"
  }
  object {
    data_object_type = "TABLE"
    name = "main.foo.table_C"
  }
}
```

### Expected Behavior
Object for `main.foo.table_B` is added to the share

### Actual Behavior
Apply fails with `cannot update share: Shared Table 'foo.table_C' already exists`

When explicitly providing `shared_as` both before and after adding new object, it works.

Most likely https://github.com/databricks/terraform-provider-databricks/blob/0693e4c31f4fa84f2e01d5b0f75271f4c0b46fd8/catalog/resource_share.go#L152 does not work as expected.

### Terraform and provider versions
1.10.0

### LINKED ISSUE #2240: [ISSUE] Updating the `shared_as` field for tables is not supported yet for `databricks_share` resource
The [API spec](https://docs.databricks.com/api-explorer/workspace/shares/update) is incorrect, `shared_as` is only allowed for new tables in shares at the moment. Therefore, need to fix diff calculation and patch payload

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

- Improve diff detection for `databricks_share` by selectively comparing certain fields. Also suppress sending `shared_as` for update calls.
- Migrate Create & Delete to Go SDK. Update & Get calls cannot be migrated yet, due to missing alias support.
- Update doc to call out the ordering of `object`
This closes #2240, which should closes #2167

## Tests

Also manually tested locally.

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by nkvuong** [comment]
Acceptance tests passed - failed tests are not related to shares

**Review by marekbrysa — APPROVED** [review]
I also verified that it worked for me on a real-world case.

File: catalog/resource_share.go [hunk 1]
```diff
@@ -128,19 +120,23 @@ func (si ShareInfo) resourceShareMap() map[string]SharedDataObject {
...
-func Equal(a, b SharedDataObject) bool {
-	if b.SharedAs == "" {
-		b.SharedAs = a.SharedAs
+func (sdo SharedDataObject) Equal(other SharedDataObject) bool {
+	if other.SharedAs == "" {
+		other.SharedAs = sdo.SharedAs
 	}
-	return reflect.DeepEqual(a, b)
+	//don't compare computed fields
+	other.AddedAt = sdo.AddedAt
```
Comment (nfx) [inline catalog/resource_share.go:128]: this will pollute the state of `other`, leading to unexpected behavior

**Review by nfx — CHANGES_REQUESTED** [review]

File: catalog/resource_share.go (hunk 1 above)
Comment (nkvuong) [reply catalog/resource_share.go:128]: I thought `other` is passed by value, so won't be modified outside the function?

**Comment by junwei-db** [comment]
@nkvuong Any updates on this PR?

**Review by nfx — APPROVED** [review]

**Pull request merged**